apps/api/src/services/study-rooms.ts

```diff
@@ -0,0 +1,41 @@
+import type { database } from "@packages/db";
+import { and, eq, gte, lte } from "@packages/db/drizzle";
+import { studyRoom } from "@packages/db/schema";
+
+type StudyRoomsServiceInput = {
+  location?: string;
+  capacityMin?: number;
+  capacityMax?: number;
+  isTechEnhanced?: boolean;
+};
+
+export class StudyRoomsService {
+  constructor(private readonly db: ReturnType<typeof database>) {}
+
+  async getAllStudyRooms() {
+    return this.db
+      .select()
+      .from(studyRoom)
+      .then((rows) => rows);
+  }
+
+  async getStudyRoomById(id: string) {
+    const [room] = await this.db.select().from(studyRoom).where(eq(studyRoom.id, id));
+    return room || null;
+  }
+
+  async getStudyRooms(input: StudyRoomsServiceInput) {
+    const conditions = [];
+    if (input.location) conditions.push(eq(studyRoom.location, input.location));
+    if (input.capacityMin) conditions.push(gte(studyRoom.capacity, input.capacityMin));
+    if (input.capacityMax) conditions.push(lte(studyRoom.capacity, input.capacityMax));
+    if (input.isTechEnhanced !== undefined)
+      conditions.push(eq(studyRoom.techEnhanced, input.isTechEnhanced));
+
+    return this.db
+      .select()
+      .from(studyRoom)
+      .where(and(...conditions))
+      .then((rows) => rows);
```

Comment:
Same here

Reply:
I'm not understanding how this part is unnecessary. What is favored instead of it? I get that since we shouldn't be supporting an "all" endpoint, that the earlier segment should be removed, but for this one, the conditions map what should be returned. I believe "then" can be removed if async is used, to create something like this:
   return await this.db
      .select()
      .from(studyRoom)
      .where(and(...conditions))
      
 What is the best solution?

Reply:
Sorry I should have been more clear - I meant that the `.then((rows) => rows)` part seems unnecessary because no data processing is happening, so it could just be left out.